### ISSUE TITLE
Vibration or Signal Tone alarm

### ISSUE THREAD

**Issue by GisoSchroederSAP** [corona-warn-app/cwa-wishlist#158]
Original proposal/question: "Guten Tag zusammen, zunächst danke für die digitale Unterstützung!

Vorschlag: Immer mehr Firmen bieten Armbänder als Abstandswarner an, diese geben ein Signal bei Unterschreitung des Mindestabstandes an.

Kann die App das nicht auch, in Form von Ton und Vibration am Handy. Die Kombination wäre doch ideal, beispielsweise könnten Firmen hiermit sicherstellen, dass nicht ganze Organisationsbereiche in Quarantäne gehen müssen, sondern gezielt Einzelpersonen.
"

Question regarding re-use the distance measuring to support also the feature of generating a signal ton or vibration warning in case the minimum distance is underrun (maybe with a toggle on/off switch).

**Comment by daimpi** [comment]
I think live distance measurement will face some of the same limitations as trying to count contacts. In particular, ENF doesn't seem to expose data on non-positive contacts to CWA: https://github.com/corona-warn-app/cwa-app-ios/issues/678#issuecomment-646686363

Another issue is that in order to get a somewhat reliable distance measurement you'd need to decrypt the metadata which is afaik only possible if a person shares their TEKs (which only happens after they test positive).

**Comment by GisoSchroederSAP** [comment]
I think, the proposed feature is not that closely-linked to the any positive or negative test result. In the end, the user is just asking, if the distance measurement could be re-used to get warned in case you are too close to someone else .

**Comment by daimpi** [comment]
Yes, but how can CWA do this if it cannot even access the RPIs from ppl who are non-positive because this data is hidden by the ENF?

And even if you could somehow access the data: ENF only scans approximately every 5 min [for a few seconds to record beacons](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#limitations). This would leave gaps which are way too large for any type of real-time distance measurement.

**Comment by GisoSchroederSAP** [comment]
@daimpi , my simple understanding of the request is: A new feature that is purely based on the BLE technology (signal strength) to either bring a signal and/or vibration in case the current distance is lower than a (predefined) minimal distance. 
There is no key matching involved at all - as far as I understand, just measuring the signal strength and calculating the distance out of if would be sufficient. 
Those parameters anyway will be retrieved during the "common usage" of the App, way before any test result comes in. So, IMHO, the link to https://github.com/corona-warn-app/cwa-app-ios/issues/678#issuecomment-646686363 does not work for this issue.

I'll take it again to the DEV team and architects. If it is technically possible to achieve this function, we will try to put this into the DEV planning, according to the priority.

**Comment by daimpi** [comment]
@GisoSchroederSAP maybe we're talking past each other here, lemme try to clarify a bit 🙂. 

Is the system you're envisioning one 
1. in which CWA itself starts sending and receiving BLE signals (outside of ENF) or
2. in which you try using the BLE signals & recordings from ENF to determine distance?

My responses so far were assuming that you're trying to take the latter (2) approach, as my current understanding is that former approach (1) is disallowed by Google/Apple for apps utilizing ENF (cf. the discussion in this issue [here](https://github.com/corona-warn-app/cwa-wishlist/issues/100) from a while back).

**Comment by GisoSchroederSAP** [comment]
@daimpi , that's true. Subsequently, this leads to the answer, that with the current design decissions this app will not be suitable to support any kind of distance warnung by signal of vibration, as the distance approximation is _**only**_ done for relevant contacts with positive-tested persons.
